Os/Posix/test/ut/PosixFileTests.cpp

```diff
@@ -0,0 +1,144 @@
+    std::shared_ptr<std::string> pointer(new std::string(full_buffer), std::default_delete<std::string>());
+    return pointer;
+}
+
+//! Clean-up the files created during this test.
+//!
+void cleanup(int signal) {
+    // Ensure the test files are removed only when the test was run
+    for (const auto& val : FILES) {
+        if (check_permissions(val->c_str(), F_OK)) {
+            ::unlink(val->c_str());
+        }
+    }
+    FILES.clear();
+}
+
+//! Set up for the test ensures that the test can run at all
+//!
+void setUp(bool requires_io) {
+    std::shared_ptr<std::string> non_random_filename = get_test_filename(false);
+    int result = mkdir(BASE_PATH, 0777);
+    // Check that we could make the directory for test files
+    if (result != 0 && errno != EEXIST) {
+        GTEST_SKIP() << "Cannot make directory for test files: " << strerror(errno);
+    }
+    // IO required and test file exists then skip
+    else if (check_permissions(non_random_filename->c_str(), F_OK)) {
+        GTEST_SKIP() << "Test file exists: " << non_random_filename->c_str();
+    }
+    // IO required and cannot read/write to BASE_PATH then skip
+    else if (requires_io && not check_permissions(BASE_PATH, R_OK & W_OK)) {
+        GTEST_SKIP() << "Cannot read/write in directory: " << BASE_PATH;
+    }
+    int signals[] = {SIGQUIT, SIGABRT, SIGTERM, SIGINT, SIGHUP};
+    for (unsigned long i = 0; i < FW_NUM_ARRAY_ELEMENTS(signals); i++) {
+        // Could not register signal handler
+        if (signal(SIGQUIT, cleanup) == SIG_ERR) {
+            GTEST_SKIP() << "Cannot register signal handler for cleanup";
+        }
+    }
+}
+
+//! Tear down for the tests cleans up the test file used
+//!
+void tearDown() {
+    cleanup(0);
+}
+
+class PosixTester : public Tester {
+    //! Check if the test file exists.
+    //! \return true if it exists, false otherwise.
+    //!
+    bool exists(const std::string& filename) const override {
+        bool exits =  check_permissions(filename.c_str(), F_OK);
+        return exits;
+    }
+
+    //! Get a filename, randomly if random is true, otherwise use a basic filename.
+    //! \param random: true if filename should be random, false if predictable
+    //! \return: filename to use for testing
+    //!
+    std::shared_ptr<const std::string> get_filename(bool random) const override {
+        U32 pick = STest::Pick::lowerUpper(0, MAX_FILES);
+        if (random && pick < FILES.size()) {
+            return FILES[pick];
+        }
+        std::shared_ptr<const std::string> filename = get_test_filename(random);
+        FILES.push_back(filename);
+        return filename;
+    }
+
+    //! Posix tester is fully functional
+    //! \return true
+    //!
+    bool functional() const override{
+        return true;
+    }
+
+};
+
+std::unique_ptr<Os::Test::File::Tester> get_tester_implementation() {
+    return std::unique_ptr<Os::Test::File::Tester>(new Os::Test::File::PosixTester());
```

Comment:
I think this should be std::make_unique in modern C++ style.

Reply:
std::make_unique is a C++14 feature, while we are at C++11.  I found this information here:

https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique

Reply:
OK, sounds good.